java/datastore/src/test/java/com/google/datastore/v1/client/RemoteRpcTest.java

```diff
@@ -122,54 +119,48 @@ public void testPlainTextException() {
 
     InputStream is = rpc.call("beginTransaction", BeginTransactionResponse.getDefaultInstance());
-    BeginTransactionResponse parsedResp = BeginTransactionResponse.parseFrom(is);
+    BeginTransactionResponse parsedResponse = BeginTransactionResponse.parseFrom(is);
     is.close();
 
-    assertEquals(resp, parsedResp);
-    assertFalse(is instanceof GzipFixingInputStream);
+    assertEquals(response, parsedResponse);
   }
 
   @Test
   public void testGzipHack_Gzip() throws Exception {
-    BeginTransactionResponse resp = newBeginTransactionResp();
-    InjectedTestValues injectedTestValues = new InjectedTestValues(gzip(resp), new byte[1], true);
+    BeginTransactionResponse response = newBeginTransactionResp();
+    InjectedTestValues injectedTestValues = new InjectedTestValues(gzip(response), new byte[1], true);
     RemoteRpc rpc = newRemoteRpc(injectedTestValues);
 
     InputStream is = rpc.call("beginTransaction", BeginTransactionResponse.getDefaultInstance());
-    BeginTransactionResponse parsedResp = BeginTransactionResponse.parseFrom(is);
+    BeginTransactionResponse parsedResponse = BeginTransactionResponse.parseFrom(is);
     is.close();
 
-    assertEquals(resp, parsedResp);
-    assertTrue(is instanceof GzipFixingInputStream);
-    assertEquals(1, ((GzipFixingInputStream) is).callsToRead);
+    assertEquals(response, parsedResponse);
     // Check that the underlying stream is exhausted.
     assertEquals(-1, injectedTestValues.inputStream.read());
   }
 
   @Test
   public void testGzipHack_GzipTooManyExtraBytes() throws Exception {
-    BeginTransactionResponse resp = newBeginTransactionResp();
+    BeginTransactionResponse response = newBeginTransactionResp();
     // NOTE(eddavisson): We might expect 101 extra bytes to be enough that the underlying input
     // stream is not exhausted, but this is not the case (likely due to a buffer somewhere). 1000
-    // extra bytes seems to be enough. We check the value of callsToRead directly to make sure
-    // we eventually stopped trying to consume the underlying stream.
+    // extra bytes seems to be enough.
     InjectedTestValues injectedTestValues =
-        new InjectedTestValues(gzip(resp), new byte[1000], true);
+        new InjectedTestValues(gzip(response), new byte[1000], true);
     RemoteRpc rpc = newRemoteRpc(injectedTestValues);
 
     InputStream is = rpc.call("beginTransaction", BeginTransactionResponse.getDefaultInstance());
-    BeginTransactionResponse parsedResp = BeginTransactionResponse.parseFrom(is);
+    BeginTransactionResponse parsedResponse = BeginTransactionResponse.parseFrom(is);
     is.close();
 
-    assertEquals(resp, parsedResp);
-    assertTrue(is instanceof GzipFixingInputStream);
-    assertEquals(100, ((GzipFixingInputStream) is).callsToRead);
+    assertEquals(response, parsedResponse);
     // Check that the underlying stream is _not_ exhausted.
     assertNotEquals(-1, injectedTestValues.inputStream.read());
```

Comment:
I would expect this assertion to fail because the test depends on MAX_BYTES_TO_CONSUME which is being deleted. I think it should be fine to just keep the one test that checks that gzipped streams are now fully-consumed with the new http client version.